src/managers/backup.py

```diff
@@ -136,6 +137,26 @@ def create_backup(self) -> BackupMetadata:
 
         return metadata
 
+    def list_backups(self) -> list[BackupMetadata]:
```

Comment:
**question** should we provide a way to paginate trhough the backups?

Reply:
No other charm provides this feature, so I have not considered it for this PR. We can think about it and add an addendum to the spec docs, though

Reply:
I suppose it is related to the discussion [here](https://github.com/canonical/zookeeper-k8s-operator/pull/106#discussion_r1763434600). Let's check the temperature with the rest of the team